oml/utils/audios.py

```diff
@@ -0,0 +1,107 @@
+import base64
+from io import BytesIO
+from typing import Union
+
+import matplotlib.pyplot as plt
+import numpy as np
+import torchaudio
+from numpy.typing import NDArray
+from PIL import Image
+from torch import FloatTensor
+
+from oml.const import BLACK, TColor
+
+
+def _visualize_audio(
+    spec_repr: FloatTensor, color: TColor = BLACK, draw_bbox: bool = True, return_b64: bool = False
+) -> Union[NDArray[np.uint8], str]:
+    """
+    Internal function to visualize an audio spectrogram.
+
+    Args:
+        spec_repr: The spectrogram representation as a FloatTensor.
+        color: The color of the bounding box.
+        draw_bbox: Whether to draw a bounding box around the spectrogram.
+        return_b64: Whether to return the image as a base64 string.
+
+    Returns:
+        If return_b64 is False, returns the image as a NumPy array.
+        If return_b64 is True, returns the image as a base64 string.
+    """
+    fig, ax = plt.subplots(figsize=(2.56, 2.56), dpi=100)
+
+    # actual image
+    ax.imshow(spec_repr, aspect="auto", origin="lower")
+
+    # bbox and axes
+    if draw_bbox:
+        frame_thickness = 5
+        for axis in ["top", "bottom", "left", "right"]:
+            ax.spines[axis].set_linewidth(frame_thickness)
+            ax.spines[axis].set_edgecolor(
+                [c / 255 for c in color]
+            )  # TODO: change all colors from arrays to strings in vis functions
+    ax.set_xticks([])
+    ax.set_yticks([])
+
+    # drawing
+    fig.canvas.draw()
+    buf = BytesIO()
+    plt.savefig(buf, format="png", bbox_inches="tight")
+    buf.seek(0)
+    img: Union[NDArray[np.uint8], str] = (
+        base64.b64encode(buf.getvalue()).decode("ascii") if return_b64 else np.array(Image.open(buf), dtype=np.uint8)
+    )
+    plt.close(fig)
+    return img
+
+
+def visualize_audio(spec_repr: FloatTensor, color: TColor = BLACK, draw_bbox: bool = True) -> NDArray[np.uint8]:
```

Comment:
`numpy 1.21` 
[link](https://numpy.org/devdocs/reference/typing.html#numpy.typing.NDArray)

Reply:
done

Reply:
I've [found](https://github.com/numpy/numpy/releases/tag/v1.21.0) the date of the release: it's June 2021. Hm, probably some users still use it. Anyway, in the rest of the repo we just use `np.ndarray`, let's do the same here in order to avoid any problems.

Reply:
what should I do if mypy goes angry on `np.ndarray`? 
```
 Missing type parameters for generic type "ndarray"
```

Reply:
probably you have a newer mypy, for CI's one it should be fine 
so just ignore that

Reply:
done